akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -222,7 +222,16 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
             ResourceFile(url, entry.getSize, entry.getTime)
           }
         } finally jar.close()
-      case _ ⇒ None
+      case _ ⇒
+        val conn = url.openConnection()
+        try {
+          conn.setUseCaches(false) // otherwise the JDK will keep the JAR file open when we close!
+          val len = conn.getContentLength
+          val lm = conn.getLastModified
+          Some(ResourceFile(url, len, lm))
+        } catch {
+          case _: Throwable ⇒ None
```

Comment:
> The wildcard catch matches the previous code in that the semantics changed from a default value of 0 for the extracted tuple of length and lastModified to a monad that required provision. The try-finally pattern of the old code had an implicit catch of all values that are ignored.

Were are the default `0` values and the implicit catch in the [old version](https://github.com/2beaucoup/akka/blob/aa8da46068795a1110ed3ce07cf648fedaf1000d/akka-http-scala/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala#L92-L99)?